api/responses/members_in_cd.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/*
+    Copyright (C) 2004-2010 Kestas J. Kuliukas / Timothy Jones
+
+	This file is part of webDiplomacy.
+
+    webDiplomacy is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    webDiplomacy is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with webDiplomacy.  If not, see <http://www.gnu.org/licenses/>.
+ */
+namespace webdiplomacy_api;
+
+defined('IN_CODE') or die('This script can not be run by itself.');
+
+require_once ('game_country.php');
+
+/**
+ * Games in CD JSON response
+ * @package webdiplomacy_api
+ */
+class CountriesInCivilDisorder {
+	/**
+	 * List of (gameID, countryID)
+	 * @var array
+	 */
+	public $value = array();
+
+	/**
+	 * Load the countries in CD;
+	 */
+	function load()
+	{
+		global $DB;
+
+		$countryTabl = $DB->sql_tabl("SELECT m.gameID, m.countryID
+                                      FROM wD_Members AS m
+                                      LEFT JOIN wD_Orders AS o ON ( o.gameID = m.gameID AND o.countryID = m.countryID )
+                                      LEFT JOIN wD_Games AS g ON ( g.id = m.gameID )
+                                      WHERE NOT o.id IS NULL

```

Comment:
This SQL query is trying to find any members (either active or CD) that have not yet submitted orders on any game that will be processed in the next 60 secs.

The missedPhase attribute is not touched, therefore that player will trigger the CD logic at the next process() (within 60 secs) and will be marked as left, therefore someone else can take its place.

If I wait until the player is marked as "Left", there is at least one turn where no orders will be submitted at all. Let me know what logic you prefer.

The model should give reasonable moves even for "Press" games, because it uses the previous orders as a proxy for relationships. I'm not sure we need to add the 'NoPress' clause, but I can add it if you want.

I'll add the 'DISTINCT' keyword and remove the 10 mins minimum phaseLimit clause.

Reply:
Oh okay I understand the confusion now. So in the past couple weeks I pushed live a change that added "ExcusedMissedTurns" to games. So each game now has 0-4 missed turns. The default is 1. If a player does not have orders entered when the turn goes to process they are charged a missed turn and the phase resets time resets to prevent the case where no orders will be submitted. If the player does not have an excuses left they are kicked into CD and the phase time resets, meaning that you'll be able to only look for people who have a status of left, and pick moves for them, without causing games to process with missing turns. Your code and this recent change should go really nicely hand in hand to make the game flow smoother for members. 

For the initial launch I would like to limit it to no press games, if there aren't enough games that meet the criteria then we should be able to setup some games with moderator"dummy" accounts taking up 1 position, force them into CD and allow the players to play against 1 of the bots in a private game to ensure nobody takes the position over. And then once we've had some successful games go through we can roll this out to press games. I'll just want a little time to put a measure in place to prevent this from happening on some press games that are apart of tournaments or other special event games. So I'll get something in place for that asap. 

Let me know if you need more details on the recent change adding missedTurns, but I tried to make sure it wouldn't conflict with your code at all.